MetaMask/metamask-extension
# Adding initial and transparent to Box component

## Explanation

Currently, `INITIAL` and `TRANSPARENT` color options for the box component don't exist or don't work. We also still allow for SECONDARY colors which are now deprecated

This is a problem because it reduces the flexibility of the Box component and is required for the new icon component. 

In order to solve this problem, this pull request adds and fixes the `INITIAL` and `TRANSPARENT` color options for the `Box` component. It also removes the deprecated SECONDARY colors.


## More Information

Fixes #15559 

See deprecated colors list in the [design tokens repo](https://metamask.github.io/design-tokens/?path=/story/colors-themecolors--light-theme-colors)

<img width="1440" alt="Screen Shot 2022-08-11 at 3 55 42 PM" src="https://user-images.githubusercontent.com/8112138/184256012-f3a0022f-8536-46e0-a076-2536b456643a.png">


## Screenshots/Screencaps

### Before

<img width="1440" alt="Screen Shot 2022-08-11 at 3 58 32 PM" src="https://user-images.githubusercontent.com/8112138/184256773-5a3ebb5d-b8ff-45c3-b087-6f542324a50a.png">
<img width="1440" alt="Screen Shot 2022-08-11 at 3 59 25 PM" src="https://user-images.githubusercontent.com/8112138/184256778-921689f7-9acc-4e67-8965-0413f9314334.png">
<img width="1440" alt="Screen Shot 2022-08-11 at 3 59 51 PM" src="https://user-images.githubusercontent.com/8112138/184256779-08108fab-e7e7-4999-a411-464cfcaef25d.png">


### After

<img width="1440" alt="Screen Shot 2022-08-11 at 4 01 59 PM" src="https://user-images.githubusercontent.com/8112138/184256807-ffa53ea4-26d3-4145-ab08-b892929ad4f6.png">
<img width="1440" alt="Screen Shot 2022-08-11 at 4 02 52 PM" src="https://user-images.githubusercontent.com/8112138/184256809-3dd2e868-6703-4937-b6f9-56f66d0fe297.png">


## Manual Testing Steps

- Go to the storybook build on this PR
- Search `Box` in the search bar
- Use the controls to check that `transparent` is working for **backgroundColor**
- Use the controls to check that `inherit` is working for **color**
- See that all the links to design system OBJECT go to the correct line positions
- See that any instances of SECONDARY has been removed

## Pre-Merge Checklist


### + If there are functional changes:
